app/app/src/main/kotlin/com/hedvig/app/feature/referrals/ui/tab/ReferralsGraph.kt

```diff
@@ -217,6 +218,72 @@ private fun ForeverScreen(
     refreshingOffset = PullRefreshDefaults.RefreshingOffset + systemBarInsetTopDp,
   )
   val incentive = uiState.incentive
+
+  val sheetState = rememberModalBottomSheetState()
+  val coroutineScope = rememberCoroutineScope()
+
+  LaunchedEffect(uiState.showEditCode) {
+    coroutineScope.launch {
+      if (uiState.showEditCode) {
+        sheetState.show()
+      } else {
+        sheetState.hide()
+      }
+    }
+  }
+
+  if (uiState.showEditCode) {
```

Comment:
I couldnt get this to animate properly, do you know what Im doing wrong?

Reply:
Yeah I think it's the same as I had to do here https://github.com/HedvigInsurance/android/blob/7e85f0db5781027e32700587c32c517454c95dac/app/feature/feature-insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurancedetail/yourinfo/YourInfoTab.kt#L57-L86

When you want it to go away, if you just remove it from composition by passing false to the `if` it will just disappear the moment the if turns false.
In the place I linked you at, when you try to dismiss it what you need to do is ask for the SheetState to dismiss, and after it's finished (hide() is suspending) then flip the flag, so it will animate out, and the moment it's already animated all the way then the if statement is flipped and it leaves composition.

Reply:
In this case where the state does come from the ViewModel however, where you can't defer flipping it like it's done in that sample.
I think that makes this a bit more involved, might be a bit harder to fix, maybe we should look into it later?

Reply:
I think that works, thanks!